src/overlays/actors/ovl_Bg_Spot16_Bombstone/z_bg_spot16_bombstone.c

```diff
@@ -21,49 +109,447 @@ const ActorInit Bg_Spot16_Bombstone_InitVars = {
+        do {
+            randomPosition.x = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.x;
+            randomPosition.y = ((Math_Rand_ZeroOne() * 5.0f) + this->actor.posRot.pos.y) + 8.0f;
+            randomPosition.z = ((Math_Rand_ZeroOne() - 0.5f) * 8.0f) + this->actor.posRot.pos.z;
+
+            randomVelocity.x = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+            randomVelocity.y = (Math_Rand_ZeroOne() * 14.0) + (fabsf(this->actor.velocity.y) * velocityYMultiplier);
+            randomVelocity.z = (Math_Rand_ZeroOne() - 0.5f) * 16.0f;
+
+            scale = D_808B6074[index] * this->actor.scale.x * 3;
+
+            func_80029E8C(globalCtx, &randomPosition, &randomVelocity, &this->actor.posRot.pos, -0x1A4, 0x31, 0xF, 0xF,
+                          0, scale, 2, 0x40, 0xA0, -1, OBJECT_BOMBIWA, &D_060009E0);
+            index += 1;
+        } while (index != ARRAY_COUNT(D_808B6074));
+    }
+}
+
+void func_808B561C(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    s32 index;
+    PosRot* posRot;
+
+    posRot = &this->actor.posRot;
+    for (index = 0; index < ARRAY_COUNT(D_808B6088); index++) {
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, 0xCD, posRot->pos.x, posRot->pos.y, posRot->pos.z, 0, 0, 0,
+                        D_808B6088[index]) == 0) {
+            return;
+        }
+    }
+}
+
+void func_808B56BC(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 sinValue;
+    s16 adjustedYawDiff;
+    s32 yawDiff;
+    s32 absYawDiff;
+
+    if (this->actor.xzDistFromLink < 130.0f && this->actor.yDistFromLink < 160.0f &&
+        this->actor.yDistFromLink >= -10.0f) {
+        yawDiff = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+        absYawDiff = (yawDiff >= 0) ? yawDiff : -yawDiff;
+
+        adjustedYawDiff = absYawDiff - 0x3FFF;
+
+        if (adjustedYawDiff > 0) {
+            sinValue = Math_Sins(adjustedYawDiff) * this->actor.xzDistFromLink;
+
+            if (sinValue >= 0.0f) {
+                player->actor.posRot.pos.x += sinValue * this->sinRotation;
+                player->actor.posRot.pos.z += sinValue * this->cosRotation;
+            } else {
+                osSyncPrintf("Error 補正出来ない(%s %d)(arg_data 0x%04x)(hosei_angY %x)\n",
+                             "../z_bg_spot16_bombstone.c", 935, this->actor.params, adjustedYawDiff);
+            }
+        }
+    }
+}
+
+void func_808B57E0(BgSpot16Bombstone* this, GlobalContext* globalCtx) {
+    EnBombf* playerHeldActor;
+    Player* player = PLAYER;
+    EnBombf* currentBomb;
+
+    if (timer > 0) {
+        timer--;
+    }
+
+    if (playerBomb != NULL) {
+        if (playerBomb->actor.update == NULL) {
+            playerBomb = NULL;
+        } else if (timer <= 0 && playerBomb->actor.posRot.pos.y < 1400.0f &&
+                   Math3D_Dist1DSq(playerBomb->actor.posRot.pos.x + 1579.0f, playerBomb->actor.posRot.pos.z + 790.0f) <
+                       160000.0f &&
```

Comment:
```suggestion
                       SQ(400.0f) &&
```
makes it more readable when comparing squared distances